server/tso/tso.go

```diff
@@ -35,12 +37,13 @@ const (
 	UpdateTimestampStep  = 50 * time.Millisecond
 	updateTimestampGuard = time.Millisecond
 	maxLogical           = int64(1 << 18)
+	maxResetGap          = 24 * time.Hour
 )
 
 // TimestampOracle is used to maintain the logic of tso.
 type TimestampOracle struct {
 	// For tso, set after pd becomes leader.
-	ts            atomic.Value
+	ts            unsafe.Pointer
```

Comment:
Is it necessary to use unsafe?

Reply:
We need CAS operator now.

Reply:
Could we use a meaningful pointer name like `*atomic.Value`? Covert it to an `unsafe.Pointer` when it is used.